exec/java-exec/src/test/java/org/apache/drill/exec/coord/zk/TestZKACL.java

```diff
@@ -0,0 +1,162 @@
+  private static  ACLProvider aclProviderDelegate;
+
+  private static CuratorFramework client;
+
+  @Before
+  public void setUp() throws Exception {
+    System.setProperty("zookeeper.authProvider.1", "org.apache.zookeeper.server.auth.SASLAuthenticationProvider");
+    String configPath = (ClassLoader.getSystemResource("zkacltest.conf")).getPath();
+    System.setProperty("java.security.auth.login.config", configPath);
+    server = new TestingServer();
+
+
+    final DrillConfig config = new DrillConfig(DrillConfig.create().withValue(ExecConstants.ZK_ACL_PROVIDER,
+            ConfigValueFactory.fromAnyRef("creator-all")
+    ).withValue(ExecConstants.ZK_APPLY_SECURE_ACL, ConfigValueFactory.fromAnyRef(true)));
+
+    final ScanResult result = ClassPathScanner.fromPrescan(config);
+    final BootStrapContext bootStrapContext =
+            new BootStrapContext(config, SystemOptionManager.createDefaultOptionDefinitions(), result);
+    aclProviderDelegate = ZKACLProviderFactory.getACLProvider(config, drillClusterPath, bootStrapContext);
+
+    server.start();
+
+    client =  CuratorFrameworkFactory.builder().
+            retryPolicy(retryPolicy).
+            connectString(server.getConnectString()).
+            aclProvider(aclProviderDelegate).
+            build();
+    client.start();
+  }
+
+  /**
+   * Test ACLs on znodes required to discover the cluster
+   */
+  // ZK libraries only supports one client instance per-machine per-server and it is cached.
+  // This test will fail when run after other ZK tests that setup the client in a way that will cause this test to fail.
+  @Ignore
+  @Test
+  public void testClusterDiscoveryPaths() {
+    try {
+      String path = PathUtils.join(drillClusterPath, cluster_config_znode);
+      client.create().creatingParentsIfNeeded().forPath(path, cluster_config_data);
+      List<ACL> remoteACLs = client.getACL().forPath(path);
+      List<ACL> desiredACLs = ((ZKACLProviderDelegate) aclProviderDelegate).aclProvider.getDrillAclForPath(drillClusterPath);
+
+      // Check the ACLs
+      for (ACL remote : remoteACLs) {
+        boolean found = false;
+        for (ACL desired : desiredACLs) {
+          // desiredACL list is READ_ACL_UNSAFE (READ, WORLD_ANYONE) + CREATOR_ALL_ACL(ALL, AUTH)
+          // AUTH in CREATOR_ALL would translate to SASL, username. Hence the replacement
+          // Note: The username("testuser1") should match the username in java.security.auth.login.config
+          found = desired.toString().equals(remote.toString().replace("sasl", "auth").replace("testuser1", ""));
+
+          if (found) { break; }
+        }
+        Assert.assertTrue(found);
+      }
+      // check if the data can be read
+      byte[] actual = client.getData().forPath(path);
+      Assert.assertArrayEquals("testClusterDiscoveryPaths data mismatch", cluster_config_data, actual);
+
+    } catch (Exception e) {
+      e.printStackTrace();
```

Comment:
Please remove here and in the code below.

Reply:
Done